edx/edx-enterprise
# ENT-3460: add default branding config object to the Customer record if null

[ENT-3460](https://openedx.atlassian.net/browse/ENT-3460)

**Description**
- Added `branding_configuration` property to `EnterpriseCustomer` to return default config object if none exists
- Added `logo` property to `EnterpriseCustomerBrandingConfiguration` to return the platform logo URL if none exists

**Preview**
![image](https://user-images.githubusercontent.com/58956521/95607088-38ee2480-0a29-11eb-817a-90ece5694123.png)

**Testing Instructions**
- Pull branch, install enterprise into platform, run migrations, ensure the learner portal is running
- Create a new Enterprise Customer from the LMS django admin with no branding configuration
- Use the UUID for the new customer to hit the customer detail endpoint e.g. http://localhost:18000/enterprise/api/v1/enterprise-customer/[uuid]
- Observe that a branding configuration is returned in the customer record despite one not existing for the customer.
- Use the slug for the customer you created to attempt logging into the learner portal (you may need to clear your cookies for this to work properly). E.g. http://localhost:8734/[slug]
- Observe the branded login screen for your new customer and how the platform logo is displayed.
- Link a test learner with the new customer and login.
- Observe the logo successfully displayed in the learner portal header and the edX branding colors in the learner portal.
- Open an LMS shell and run `./manage.py lms shell` to initialize django and enter a python shell
- Create an instance of the `EnterpriseCustomer` model with no arguments to create a customer without a branding configuration.
- Run `print(customer.branding_configuration)` and observe the default branding configuration object printed rather than an error message for `RelatedObjectDoesNotExist`

**Merge checklist:**
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.

**Post merge:**
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.
